code/API_definitions/Get DeviceLocation.yaml

```diff
@@ -0,0 +1,237 @@
+                -'^((([^:]+:){7}([^:]+))|((([^:]+:)*[^:]+)?::(([^:]+:)*[^:]+)?))(\/.+)?$'
+          - description: |
+              IPv4 address may be specified in form <address/mask> as:
+              - address - an IPv4 number in dotted-quad form 1.2.3.4. Only this exact IP number will match the flow control rule.
+              - address/mask - an IP number as above with a mask width of the form 1.2.3.4/24.
+              In this case, all IP numbers from 1.2.3.0 to 1.2.3.255 will match. The bit width MUST be valid for the IP version.
+            name: Ipv4Address
+            in : query
+            schema:
+              type: string
+              pattern: 
+                -'^((:|(0?|([1-9a-f][0-9a-f]{0,3}))):)((0?|([1-9a-f][0-9a-f]{0,3})):){0,6}(:|(0?|([1-9a-f][0-9a-f]{0,3})))(\/(([0-9])|([0-9]{2})|(1[0-1][0-9])|(12[0-8])))?$'
+                -'^((([^:]+:){7}([^:]+))|((([^:]+:)*[^:]+)?::(([^:]+:)*[^:]+)?))(\/.+)?$'
+          - description: User equipment port. Device port may be required along with IP address to identify the target device
+            name: port
+            in : query
+            schema:
+              type: integer
+              minimum: 0
+              maximum: 65535
+      responses:
+        '200':
+          description: Location retrieval result
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RetrievalLocationResponse'
+              examples: 
+                'success':
+                  summary: successful location
+                  description: The network provides device location
+                  value:
+                    latitude: 50.735851
+                    longitude: 7.10066
+                    accuracy: 2
+
+        '400':
+          $ref: '#/components/responses/Generic400'
+        '401':
+          $ref: '#/components/responses/Generic401'
+        '403':
+          $ref: '#/components/responses/Generic403'
+        '404':
+          $ref: '#/components/responses/Generic404'
+        '500':
+          $ref: '#/components/responses/Generic500'
+        '503':
+          $ref: '#/components/responses/Generic503'
+components:
+  securitySchemes:
+    oAuth2ClientCredentials:
+      type: oauth2
+      flows:
+        clientCredentials:
+          tokenUrl: '{tokenUrl}'
+          scopes: {}
+#    BasicAuth:
+#      type: http
+#      scheme: basic
+#    apiKey:
+#      type: apiKey
+#      description: API key to authorize requests
+#      name: apikey
+#      in: query
+    three_legged:
+      type: oauth2
+      flows:
+        authorizationCode:
+          authorizationUrl: https://auth.example.com/authorize
+          tokenUrl: https://auth.example.com/token
+          scopes:
+            device-location-provide: Provide device location
+  schemas:
+    RetrievalLocationResponse:
+      type: object
+      required:
+        - latitude
+        - longitude
+        - accuracy
+      properties:

```

Comment:
This is the most obvious response. Not sure if in some cases a different Area other than Circle would make sense, if network can return a different geometry.

Reply:
Same for me. I guess circle is well enough for first release. 

Let's wait for UC requiring other Area definition and in this case we'll implement the `oneOf `model.

Reply:
Point is probably good as a starting point. There is however a potential complexity in providing only a point. Depending on the southbound system a request for location might come in another format, such as a polygon,  which then  needs to be translated into a point with an appropriate accuracy.

Reply:
Shouldn't we add a time info of this location data? I think this will be needed sooner or later, since it can be quite important for the clients.

Reply:
@akoshunyadi good question - For me it depends on if we want the API to provides current location or not last known location (associated to an age info).  This is tricky because depending on the business case one or the other could be required. 
I was thinking to keep thing simple and just provide current location or 404 if not able to provide but open to discuss this (add for example an `ageOfLocation `attribute in the response)

@jlurien WDYT?